src/test/commands.datasource.entry.test.ts

```diff
@@ -0,0 +1,743 @@
+			});
+			const eValue: string = 'NEWVALUE';
+			createDSEntryCommand.handleDataSourceEntryCreation(dvConfig, dsConfig, f, undefined, eValue)
+				.then( (created) => {
+					if (created) {
+						done(new Error('Execution of the Create DataSource Entry command returned true, but should not.'));
+					} else {
+						done();
+					}
+				})
+				.catch( (err) => {
+					done(err);
+				});
+		});
+
+		it('should not create a datasource entry inside a datasource when handing over invalid entry value', (done) => {
+			const dvConfig: IDVConfig = utils.loadModelFromFile(f);
+			const dsConfig: IDataSourceConfig = {
+				name: dsName,
+				type: mongoDSType,
+				entries: new Map()
+			};
+			dvConfig.spec.env.forEach( (element: IEnv) => {
+				const key: string = element.name.slice(prefix.length+1);
+				dsConfig.entries.set(key, element.value);
+			});
+			const eName: string = 'NEWKEY';
+			createDSEntryCommand.handleDataSourceEntryCreation(dvConfig, dsConfig, f, eName, undefined)
+				.then( (created) => {
+					if (created) {
+						done(new Error('Execution of the Create DataSource Entry command returned true, but should not.'));
+					} else {
+						done();
+					}
+				})
+				.catch( (err) => {
+					done(err);
+				});
+		});
+	});
+
+	context('Modify DataSource Entry', () => {
+		const eName: string = 'NEWKEY';
+		const eValue: string = 'NEWVALUE';
+
+		beforeEach( (done) => {
+			createVDBCommand.handleVDBCreation(p, name, templFolder)
+			.then( (createdVDB) => {
+				if (createdVDB) {
+					f = path.join(p, `${name}.yaml`);
+					fs.existsSync(f).should.equal(true);
+					const dvConfig: IDVConfig = utils.loadModelFromFile(f);
+					createDSCommand.handleDataSourceCreation(dsName, dsType, dvConfig, f)
+						.then( (createdDS) => {
+							if (createdDS) {
+								const dvConfig2: IDVConfig = utils.loadModelFromFile(f);
+								dvConfig2.should.deep.equal(dvConfig);
+								dvConfig2.spec.env.length.should.deep.equal(mongoTempl.entries.size);
+
+								const dsConfig: IDataSourceConfig = {
+									name: dsName,
+									type: mongoDSType,
+									entries: new Map()
+								};
+								dvConfig.spec.env.forEach( (element: IEnv) => {
+									const key: string = element.name.slice(prefix.length+1);
+									dsConfig.entries.set(key, element.value);
+								});
+								const oldLen: number = dvConfig.spec.env.length;
+								createDSEntryCommand.handleDataSourceEntryCreation(dvConfig, dsConfig, f, eName, eValue)
+									.then( (created) => {
+										if (created) {
+											dvConfig.spec.env.length.should.equal(oldLen+1);
+											should.exist(dvConfig.spec.env.find( (element) => {
+												return element.name === utils.generateFullDataSourceConfigEntryKey(dsConfig, eName) && element.value === eValue;
+											}));
+											done();
+										} else {
+											done(new Error('Execution of the Create DataSource Entry command returned false'));
+										}
+									})
```

Comment:
Sonar is mentioning that this inlined function is repeated several times in the file. Would be better to factorize it.